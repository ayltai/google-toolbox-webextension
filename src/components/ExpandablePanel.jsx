import { Button, makeStyles } from '@material-ui/core';
import { ExpandMore, Settings } from '@material-ui/icons';
import PropTypes from 'prop-types';
import React from 'react';
import browser from 'webextension-polyfill';

export const ExpandablePanel = props => {
    const useStyles = makeStyles(theme => ({
        container : {
            display        : 'flex',
            flexWrap       : 'wrap',
            overflow       : 'hidden',
            justifyContent : 'flex-start',
        },
        fluid     : {
            flex : 1,
        },
        button    : {
            margin : theme.spacing(2),
            width  : '100%',
        },
    }));

    const classes = useStyles();

    const [ expanded, setExpanded, ] = React.useState(false);

    return (
        <div className={classes.container}>
            {props.children}
            {!expanded && (
                <Button
                    className={classes.button}
                    color='default'
                    variant='outlined'
                    startIcon={<ExpandMore />}
                    onClick={() => {
                        setExpanded(true);

                        if (props.onExpand) props.onExpand();
                    }}>
                    More
                </Button>
            )}
            {expanded && (
                <Button
                    className={classes.button}
                    color='default'
                    variant='outlined'
                    startIcon={<Settings />}
                    onClick={() => {
                        browser.runtime.openOptionsPage();

                        window.close();
                    }}>
                    Preferences
                </Button>
            )}
        </div>
    );
};

ExpandablePanel.propTypes = {
    children : PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.arrayOf(PropTypes.node),
    ]),
    onExpand : PropTypes.func,
};
