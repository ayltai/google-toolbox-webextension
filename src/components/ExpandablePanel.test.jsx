import { Button } from '@material-ui/core';
import { mount, shallow, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import browser from 'sinon-chrome/webextensions';

import { ExpandablePanel } from './ExpandablePanel';

window.browser = browser;

describe('<ExpandablePanel />', () => {
    it('renders correctly', () => {
        expect(renderer.create(<ExpandablePanel />).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(<ExpandablePanel />);
    });

    it('triggers onExpand when clicked', () => {
        const onExpand = jest.fn();

        shallow(<ExpandablePanel onExpand={onExpand} />)
            .find(Button)
            .simulate('click');

        expect(onExpand.mock.calls.length).toEqual(1);
    });
});
