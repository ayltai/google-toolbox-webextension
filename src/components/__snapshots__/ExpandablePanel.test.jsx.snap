// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandablePanel /> renders correctly 1`] = `
<div
  className="makeStyles-container-1"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-3"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      More
    </span>
  </button>
</div>
`;
